sunpy/time/time.py

```diff
@@ -142,67 +137,64 @@ def convert_time(time_string, **kwargs):
 
 @convert_time.register(pandas.Timestamp)
 def convert_time_pandasTimestamp(time_string, **kwargs):
-    return time_string.to_pydatetime()
+    return ap.Time(time_string)
 
 
 @convert_time.register(pandas.Series)
 def convert_time_pandasSeries(time_string, **kwargs):
-    if 'datetime64' in str(time_string.dtype):
-        return np.array([dt.to_pydatetime() for dt in time_string])
-    else:
-        convert_time.dispatch(object)(time_string, **kwargs)
+    return ap.Time(time_string.tolist())
 
 
 @convert_time.register(pandas.DatetimeIndex)
 def convert_time_pandasDatetimeIndex(time_string, **kwargs):
-    return time_string._mpl_repr()
+    return ap.Time(time_string.tolist())
 
 
 @convert_time.register(datetime)
 def convert_time_datetime(time_string, **kwargs):
-    return time_string
+    return ap.Time(time_string)
```

Comment:
I see that there are a few of these dispatch functions that are just `ap.Time(time_string)` we should just define this once and have multiple decorators on it, i.e.

```python
@convert_time.register(datetime)
@convert_time.register(pandas.Timestamp)
@convert_time.register(astropy.time.Time)
 def convert_time_constructor(time_string, **kwargs):
    return ap.Time(time_string)
```